q2_quality_control/_threshold_graph/_visualizer.py

```diff
@@ -7,84 +7,169 @@
-    subset_key_arr = []
-    contam_val_arr = []
-    true_val_arr = []
-    unknown_val_arr = []
-    percent_val_arr = []
-    gray_lab_arr = []
-    red_lab_arr = []
-    blue_lab_arr = []
+    # TODO: Replace rep_seq_indicator to a boolean value
+    # Sets rep seq flags if rep_seq indciator is >1 then
+    # the sequences are printed in table otherwise no sequences are
+    # printed and temp lists
+    rep_seq_indicator = True
+
+    # initializes arrays to pass data to the html
+    image_paths_arr = []  # array for image paths for render on template
+    # (length 1 when running base decontam-score-viz)
+    subset_key_arr = []  # array for table subset id when runnning batches
+    # (length 1 when running base decontam-score-viz)
+    contam_val_arr = []  # contaminant features count
+    # (length 1 when running base decontam-score-viz)
+    true_val_arr = []  # non-contaminant features count
+    # (length 1 when running base decontam-score-viz)
+    unknown_val_arr = []  # NA features count
+    # (length 1 when running base decontam-score-viz)
+    percent_val_arr = []  # % contaminant features
+    # (length 1 when running base decontam-score-viz)
+    red_lab_arr = []  # contaminant feature/read label
+    # (length 1 when running base decontam-score-viz)
+    blue_lab_arr = []  # non-contaminant feature/read label
+    # (length 1 when running base decontam-score-viz)
+    data_arr = []  # information for sequencing table
+    # (only render in base decontam-score-viz)
+    true_fasta_dest = []  # destination for true/nan seq fastas
+    # (only render if rep-seqs not None)
+    contam_fasta_dest = []  # destination for contaminant feature fastas
+    # (only render if rep-seqs not None)
+    sorted_key_arr = []  # sorted feature ids
+    # (only render in base decontam-score-viz)
+    feature_or_read_arr = []  # feature or read label for graph rendering
+    # (length 1 when running base decontam-score-viz)
 
+    # iterates through tables and keys of ASV tables and decontam score tables
     for key in table_dict.keys():
         table = table_dict[key]
         decontam_scores = decontam_scores_dict[key]
-        df = decontam_scores.to_dataframe()
-        if df['p'].isna().all():
+        if decontam_scores['p'].isna().all():
             raise ValueError("No p-values exist for the data provided.")
 
+        # pull out relevant data from objs
         read_nums = table.sum(axis='rows')
-
-        p_vals = df['p'].dropna()
+        p_vals = decontam_scores['p'].dropna()
         filt_read_nums = read_nums[p_vals.index]
 
+        # start ASV contaminant differentiation
         contams = (p_vals < threshold)
 
+        # parses index for contaminant identification
+        true_indices = contams[~contams].index
+        contam_indices = contams[contams].index
+        nan_indices = decontam_scores[
+            decontam_scores['p'].isna()].index.tolist()
+
+        # if rep reqs are not found then dummy variables
```

Comment:
I think this bit about dummy variables is no longer relevant

Reply:
removed and added a bit more about the logic of the replacement code 

addressed in [b466ed9](https://github.com/qiime2/q2-quality-control/pull/82/commits/b466ed97b7fe9af561c1e13576a6c52a19c79fa6)